templates/fact.yaml.erb

```diff
@@ -1,3 +1,7 @@
 # custom fact <%= @title %>
----
-<%= @title %>: "<%= @value %>"
+<%- require 'yaml' -%>
+<%- if RUBY_VERSION =~ /^1\.8\./ -%>
+<%= YAML::dump(@facter_data) %>
+<%- else -%>
+<%= @facter_data.to_yaml(indentation:2, canonical:false, line_width:-1) %>
```

Comment:
I just tried using this code to output a text fact surrounded in single quotes and it remove the quotes around the text and didn't add the fact to the node.
It seems to work for arrays.
Can you confirm this performs properly with non structured facts?